### PR TITLE
fixed chap 14.2/3 wording/added concepts

### PR DESCRIPTION
**Pull request by zhunation**
I believe that this should contain all my changes including grammar from the previous PR, so I would only need to merge this branch

### CODE REVIEW

**Comment by zhunation** [comment]
Git was really annoying, especially with all the unicode/ windows formatting issues. I ended up just putting both the grammar and concept changes in one PR, or else it would have been really complicated merging them separately.

File: notebooks/ch14/bias_risk.md [hunk 1]
```diff
@@ -58,13 +58,13 @@ L(\hat{\theta}, X, y)
 \end{aligned}
 $$
 
-As before, we use $ \hat{\theta} $ as our vector of model parameters, $ x $ as a vector containing a row of a data matrix $ X $, and $ y $ as our vector of observed values to predict.
+As before, we use $ \hat{\theta} $ as our vector of model parameters, $ x $ as a vector containing a row of a data matrix $ X $, and $ y $ as our vector of observed values to predict. $ X_i $is the $i$'th row of $ X $ and $ y_i $ is the $i$'th entry of y.
```
Comment (SamLau95) [inline notebooks/ch14/bias_risk.md]: double check space; We use $ X_i $ to represent the $i$'th row of $ X $ and $ y_i $ to represent the $i$'th entry of $y$.

File: notebooks/ch14/bias_risk.md [hunk 2]
```diff
@@ -82,6 +82,31 @@ The expression above abstracts over the specific loss function; regardless of th
 
 By minimizing the cost, we select the model parameters that best fit our observed dataset. Thus far, we have refrained from making statements about the population that generated the dataset. In reality, however, we are quite interested in making good predictions on the entire population, not just our data that we have already seen.
 
+## Example: Cost Minimization
```
Comment (SamLau95) [inline notebooks/ch14/bias_risk.md]: Move this to ch12

File: notebooks/ch14/bias_modeling.md [hunk 3]
```diff
@@ -204,15 +228,25 @@ Cross-validation provides a more accurate method of estimating our model error u
...
 
-Underfitting is typically caused by too much bias; overfitting is typically caused by too much variance.
+Underfitting is typically caused by too much bias; overfitting is typically caused by too much model variance.
 
-Collecting more data reduces variance. One recent trend is to select a model with low bias and high intrinsic variance (e.g. a neural network) and collect many data points so that the model variance is low enough to make accurate predictions. While effective in practice, collecting enough data for these models tends to require large amounts of time and money.
+Collecting more data reduces variance. In the extreme case that we can collect all the data for something, we would be able to create a model with zero variance- meaning that it predicts every single point correctly. One recent trend is to select a model with low bias and high intrinsic variance (e.g. a neural network) and collect many data points so that the model variance is low enough to make accurate predictions. While effective in practice, collecting enough data for these models tends to require large amounts of time and money.
```
Comment (SamLau95) [inline notebooks/ch14/bias_modeling.md]: for example, model variance of linear regression goes down as 1/n . Thus, doubling the dataset halves the model variance, and collecting many data points will cause the variance to approach 0.

File: notebooks/ch14/bias_modeling.md [hunk 4]
```diff
@@ -204,15 +228,25 @@ Cross-validation provides a more accurate method of estimating our model error u
...
 
 Collecting more data reduces bias if the model can fit the population process exactly. If the model is inherently incapable of modeling the population (as in the example above), even infinite data cannot get rid of model bias.
 
 Adding a useful feature to the data, such as a quadratic feature when the underlying process is quadratic, reduces bias. Adding a useless feature rarely increases bias.
 
-Adding a feature, whether useful or not, typically increases model variance. In order to increase a model's prediction accuracy, a new feature should decrease bias more than it increases variance.
+Adding a feature, whether useful or not, typically increases model variance. This is due to the fact that more features potentially will overfit the model:causing the model to be too specific for the training data, and not allow the model to generalize to potential unseen data. In order to increase a model's prediction accuracy, a new feature should decrease bias more than it increases variance.
```
Comment (SamLau95) [inline notebooks/ch14/bias_modeling.md]: This is due to the fact that more features potentially will overfit the model:causing the model to be too specific for the training data, and not allow the model to generalize to potential unseen data.

to 

...increases model variance because each new feature adds a new model parameter, increasing the number of possible models (...).

File: notebooks/ch14/bias_modeling.md [hunk 5]
```diff
@@ -204,15 +228,25 @@ Cross-validation provides a more accurate method of estimating our model error u
...
 
 Adding a useful feature to the data, such as a quadratic feature when the underlying process is quadratic, reduces bias. Adding a useless feature rarely increases bias.
 
-Adding a feature, whether useful or not, typically increases model variance. In order to increase a model's prediction accuracy, a new feature should decrease bias more than it increases variance.
+Adding a feature, whether useful or not, typically increases model variance. This is due to the fact that more features potentially will overfit the model:causing the model to be too specific for the training data, and not allow the model to generalize to potential unseen data. In order to increase a model's prediction accuracy, a new feature should decrease bias more than it increases variance. 
+
+A lack of features will typically increase bias, and cause underfitting. Without the necessary amount of features, one won't be able to create a model that captures the underlying trend of the data. This leads to high bias because the model will rarely be able to predict the correct value, even for datapoints that we already know.
```
Comment (SamLau95) [inline notebooks/ch14/bias_modeling.md]: Removing features will typically increase bias and can cause underfitting. For example, if the underlying data come from a quadratic curve but we only use a simple linear model, then (...).

File: notebooks/ch14/bias_modeling.md [hunk 6]
```diff
@@ -204,15 +228,25 @@ Cross-validation provides a more accurate method of estimating our model error u
...
 
 Adding a useful feature to the data, such as a quadratic feature when the underlying process is quadratic, reduces bias. Adding a useless feature rarely increases bias.
 
-Adding a feature, whether useful or not, typically increases model variance. In order to increase a model's prediction accuracy, a new feature should decrease bias more than it increases variance.
+Adding a feature, whether useful or not, typically increases model variance. This is due to the fact that more features potentially will overfit the model:causing the model to be too specific for the training data, and not allow the model to generalize to potential unseen data. In order to increase a model's prediction accuracy, a new feature should decrease bias more than it increases variance. 
+
+A lack of features will typically increase bias, and cause underfitting. Without the necessary amount of features, one won't be able to create a model that captures the underlying trend of the data. This leads to high bias because the model will rarely be able to predict the correct value, even for datapoints that we already know. 
+
+The plot below is helpful in summarizing the takeaways mentioned above:
```
Comment (SamLau95) [inline notebooks/ch14/bias_modeling.md]: add short explanation of plot.

File: notebooks/ch14/bias_modeling.md [hunk 7]
```diff
@@ -229,15 +230,28 @@ Cross-validation provides a more accurate method of estimating our model error u
...
+
+Removing features will typically increase bias and can cause underfitting. For example if the underlying data comes from a quadratic curve, removing the quadratic feature and only using a simple linear model won't capture the trend of the data. 
+
+The plot below is helpful in summarizing the takeaways mentioned above:
+
+![bias_modeling_bias_var_plot.png](https://raw.githubusercontent.com/DS-100/textbook/master/assets/bias_modeling_bias_var_plot.png)
+
+As model complexity increases, we will be able to predict better values for our training set, but not extrapolate well to the test set. By decreasing model complexity, the bias will be higher for the training set, but there will be less model variance.
+
+As we add more data, we shift our plot to the right and down, reducing bias and variance:
```
Comment (SamLau95) [inline notebooks/ch14/bias_modeling.md]: We don't "shift a plot". Instead, we shift the curves on the plot.

> In the plot below, we observe that adding more data effectively moves the bias and variance curves down and to the right. This not only reduces both bias and variance but also allows us to use more complex models to achieve low test set error.

File: notebooks/ch14/bias_modeling.md [hunk 8]
```diff
@@ -229,15 +230,28 @@ Cross-validation provides a more accurate method of estimating our model error u
...
 
 Collecting more data reduces bias if the model can fit the population process exactly. If the model is inherently incapable of modeling the population (as in the example above), even infinite data cannot get rid of model bias.
 
 Adding a useful feature to the data, such as a quadratic feature when the underlying process is quadratic, reduces bias. Adding a useless feature rarely increases bias.
 
-Adding a feature, whether useful or not, typically increases model variance. In order to increase a model's prediction accuracy, a new feature should decrease bias more than it increases variance.
+Adding a feature, whether useful or not, typically increases model variance. This does so because each new feature addas a a new model parameter, increasing the total number of models. In order to increase a model's prediction accuracy, a new feature should decrease bias more than it increases variance. 
+
+Removing features will typically increase bias and can cause underfitting. For example if the underlying data comes from a quadratic curve, removing the quadratic feature and only using a simple linear model won't capture the trend of the data.
```
[the remark this answers is not in the document]
Comment (zhunation) [reply notebooks/ch14/bias_modeling.md]: A lot of these things are just taken straight from other comments you wrote. Although it sounded off, I kept it as is "Removing features will typically increase bias and can cause underfitting. For example, if the underlying data come from a quadratic curve but we only use a simple linear model, then". Next time, I'll stick to my guts and change the wording to be correct.

File: notebooks/ch14/bias_modeling.md [hunk 9]
```diff
@@ -229,15 +230,28 @@ Cross-validation provides a more accurate method of estimating our model error u
...
 
-Underfitting is typically caused by too much bias; overfitting is typically caused by too much variance.
+Underfitting is typically caused by too much bias; overfitting is typically caused by too much model variance.
 
-Collecting more data reduces variance. One recent trend is to select a model with low bias and high intrinsic variance (e.g. a neural network) and collect many data points so that the model variance is low enough to make accurate predictions. While effective in practice, collecting enough data for these models tends to require large amounts of time and money.
+Collecting more data reduces variance. For example, the model variance of linear regression goes down by a factor of $ 1
+/n $, where $ n $ is the number of data points. Thus, doubling the dataset halves the model variance, and collecting many data points will cause the variance to approach 0. One rece end is to select a model with low bias and high intrinsic variance (e.g. a neural network) and collect many data points so that the model variance is low enough to make accurate predictions. While effective in practice, collecting enough data for these models tends to require large amounts of time and money.
```
[the remark this answers is not in the document]
Comment (zhunation) [reply notebooks/ch14/bias_modeling.md]: I did write that 
>Thus, doubling the dataset halves the model variance

Typo will be fixed

File: notebooks/ch14/bias_modeling.md [hunk 10]
```diff
@@ -229,15 +230,28 @@ Cross-validation provides a more accurate method of estimating our model error u
...
 
 Collecting more data reduces bias if the model can fit the population process exactly. If the model is inherently incapable of modeling the population (as in the example above), even infinite data cannot get rid of model bias.
 
 Adding a useful feature to the data, such as a quadratic feature when the underlying process is quadratic, reduces bias. Adding a useless feature rarely increases bias.
 
-Adding a feature, whether useful or not, typically increases model variance. In order to increase a model's prediction accuracy, a new feature should decrease bias more than it increases variance.
+Adding a feature, whether useful or not, typically increases model variance. This does so because each new feature addas a a new model parameter, increasing the total number of models. In order to increase a model's prediction accuracy, a new feature should decrease bias more than it increases variance.
```
[the remark this answers is not in the document]
Comment (zhunation) [reply notebooks/ch14/bias_modeling.md]: >increases model variance because each new feature adds a new model parameter, increasing the number of possible models 

Comes from what you wrote.
Doesn't adding an additional parameter create new types of models though?   If your model is $\theta_1 *x$ and you add a quadratic parameter $ x^2$, isn't $\theta_1 *x + theta_2 *x^2$ a new model?

File: notebooks/ch14/bias_modeling.md (hunk 7 above)
Comment (zhunation) [reply notebooks/ch14/bias_modeling.md]: will be fixed

File: notebooks/ch14/bias_modeling.md [hunk 11]
```diff
@@ -229,15 +230,28 @@ Cross-validation provides a more accurate method of estimating our model error u
...
 
 Adding a useful feature to the data, such as a quadratic feature when the underlying process is quadratic, reduces bias. Adding a useless feature rarely increases bias.
 
-Adding a feature, whether useful or not, typically increases model variance. In order to increase a model's prediction accuracy, a new feature should decrease bias more than it increases variance.
+Adding a feature, whether useful or not, typically increases model variance. This does so because each new feature addas a a new model parameter, increasing the total number of models. In order to increase a model's prediction accuracy, a new feature should decrease bias more than it increases variance. 
+
+Removing features will typically increase bias and can cause underfitting. For example if the underlying data comes from a quadratic curve, removing the quadratic feature and only using a simple linear model won't capture the trend of the data. 
+
+The plot below is helpful in summarizing the takeaways mentioned above:
```
[the remark this answers is not in the document]
Comment (zhunation) [reply notebooks/ch14/bias_modeling.md]: will be fixed

File: notebooks/ch14/bias_modeling.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (SamLau95) [reply notebooks/ch14/bias_modeling.md]: Yeah, I know that part came from what I wrote. Sorry about that! I was in a rush when I first wrote the change.

I agree that those are two different models. However, "increasing the total number of models" suggests that there used to be X total models in the universe and now we've increased the total number of models from X to X + 1. In reality, adding a new parameter to a model gives the model more possible combinations of parameters which increases model variance.

File: notebooks/ch14/bias_modeling.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (SamLau95) [reply notebooks/ch14/bias_modeling.md]: Yup, sorry about that. A lot of stuff I wrote in a rush needed rewording!

**Comment by SamLau95** [comment]
Thanks for responding! And again, I know a lot of the changes were my bad. Thanks for pointing those out.

**Comment by zhunation** [comment]
@SamLau95 requested changes were made

File: notebooks/ch14/bias_modeling.md [hunk 12]
```diff
@@ -75,7 +75,7 @@ R(f_\hat{\theta}) = (\mathbb{E}[f_\hat{\theta}(z)] - f_\theta(z))^2 + \text{Var}
 \end{aligned}
 $$
 
-The first term in this expression, $ (\mathbb{E}[f_\hat{\theta}(z)] - f_\theta(z))^2 $, is a mathematical expression for the bias of the model. (Technically, this term represents the biased squared, $\text{bias}^2$). The bias is equal to zero if in the long run our choice of model $ f_\hat{\theta}(z) $ predicts the same outcomes produced by the population process $ f_\theta(z) $. The bias is high if our choice of model makes poor predictions of the population process even when we have the entire population as our dataset.
+The first term in this expression, $ (\mathbb{E}[f_\hat{\theta}(z)] - f_\theta(z))^2 $, is a mathematical expression for the bias of the model. (Technically, this term represents the biased squared, $\text{bias}^2$.) The bias is equal to zero if in the long run our choice of model $ f_\hat{\theta}(z) $ predicts the same outcomes produced by the population process $ f_\theta(z) $. The bias is high if our choice of model makes poor predictions of the population process even when we have the entire population as our dataset.
```
Comment (SamLau95) [inline notebooks/ch14/bias_modeling.md]: typo: `biased` should be `bias`.

File: notebooks/ch14/bias_modeling.md [hunk 13]
```diff
@@ -229,15 +228,28 @@ Cross-validation provides a more accurate method of estimating our model error u
...
+Adding a feature, whether useful or not, typically increases model variance since each new feature adds a parameter to the model. Generally speaking, models with many parameters have many possible combinations of parameters and therefore have higher variance than models with few parameters. In order to increase a model's prediction accuracy, a new feature should decrease bias more than it increases variance. 
+
+Removing features will typically increase bias and can cause underfitting. For example, a simple linear model has higher model bias than the same model with a quadratic feature added to it. If the data were generated from a quadratic phenomenon, the simple linear model underfits the data.
+
+In the plot below, the X-axis measures model complexity and the Y-axis measures magnitude. Notice  how as model complexity increases, model bias strictly decreases and model variance strictly increases. The test error decreases at first, but after the model reaches a certain complexity, starts increasing.
```
Comment (SamLau95) [inline notebooks/ch14/bias_modeling.md]: nit: We can move the last two words into the sentence to reduce commas:

> As we choose more complex models, the test error first decreases then increases as the increased model variance outweighs the decreased model bias.
